diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt

```diff
@@ -109,31 +112,32 @@ class DiktatRuleSetProvider(private var diktatConfigFile: String = DIKTAT_ANALYS
         yield(resolveConfigFileFromSystemProperty())
     }
 
-    /**
-     * As of _KtLint_ **0.47**, each rule is expected to have a state and is executed
-     * twice per file, and a new `Rule` instance is created per each file checked.
-     *
-     * Diktat rules have no mutable state yet and use the deprecated _KtLint_
-     * API, so we initialize them only _once_ for performance reasons and also
-     * to avoid redundant logging.
-     */
-    private val ruleSet: RuleSet by lazy {
-        log.debug("Will run $DIKTAT_RULE_SET_ID with $diktatConfigFile" +
-                " (it can be placed to the run directory or the default file from resources will be used)")
+    @Suppress(
+        "LongMethod",
+        "TOO_LONG_FUNCTION",
+    )
+    @Deprecated(
+        "Marked for removal in KtLint 0.48. See changelog or KDoc for more information.",
+    )
+    override fun get(): RuleSet {
```

Comment:
you can initialize once `configRules` and creates rules only

Reply:
Addressed with 4f01cddfaf5c731c84a9402765bd28ecc049e352